src/services/db/GitFileSystemService.ts

```diff
@@ -473,8 +499,78 @@ export default class GitFileSystemService {
       )
   }
 
-  // TODO: Update the contents of a file
-  async update() {}
+  // Update the contents of a file
+  update(
+    repoName: string,
+    filePath: string,
+    fileContent: string,
+    oldSha: string,
+    userId: SessionDataProps["isomerUserId"]
+  ): ResultAsync<string, GitFileSystemError | NotFoundError | ConflictError> {
+    return ResultAsync.fromPromise(
+      fs.promises.stat(`${EFS_VOL_PATH}/${repoName}/${filePath}`),
+      (error) => {
+        if (error instanceof Error && error.message.includes("ENOENT")) {
+          return new NotFoundError("File does not exist")
+        }
+        if (error instanceof Error) {
+          return new GitFileSystemError(error.message)
+        }
+
+        logger.error(`Error when getting ${filePath} stats: ${error}`)
+        return new GitFileSystemError("An unknown error occurred")
+      }
+    )
+      .andThen(() =>
+        this.getGitBlobHash(repoName, filePath).andThen((sha) => {
+          if (sha !== oldSha) {
+            return errAsync(
+              new ConflictError(
+                "File has been changed recently, please try again"
+              )
+            )
+          }
+          return okAsync(sha)
+        })
+      )
+      .andThen(() =>
+        ResultAsync.fromPromise(
+          fs.promises.writeFile(
```

Comment:
writing to a file and immediately reading does it give back the updated file?

do we need to make an explicit flush to the disk
https://stackoverflow.com/questions/42277513/how-to-make-fs-appendfile-method-of-node-js-flush-to-disk-right-away

Reply:
I checked around (documentation is a little sparse on this) but it seems like the `writeFile` method is a "higher level abstraction", which means that it [handles opening and closing](https://stackoverflow.com/a/21177492) of the underlying file descriptors. When the file descriptor is closed, the OS will handle the flushing of the contents to disk, so no explicit flushing to the disk would be necessary.

So in this theory, immediately reading from the file should return the updated contents. However, because we are using EFS, there might be some slight network latency, which may give a small window of opportunity where another user retrieves an older version of the file. However, I think that it might be a very small chance of it happening, and in this case the user will still be getting an older SHA, which would give them a warning about overriding changes when they try to save the file.